### PR TITLE
Revert matcher package

### PR DESCRIPTION
**Pull request by Hixie**
The new matcher package deprecates isInstanceOf which seems eggregious.

On the assumption that this will eventually become a function, I've made a function here in flutter_test and removed the "const" from calls that come from the flutter framework. I'll add a note to the changelog.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @gspencergoog

**Comment by kevmoo** [comment]
is this just for flutter's code – or will this constraint affect all users of Flutter, too?

**Comment by Hixie** [comment]
it affects any pubspec depending on flutter_test or other packages in the flutter sdk.

File: dev/bots/test/prepare_package_test.dart [hunk 1]
```diff
@@ -25,7 +25,7 @@ void main() {
         expectAsync1((List<String> commandLine) async {
           return processRunner.runProcess(commandLine);
         })(<String>['this_executable_better_not_exist_2857632534321']),
-        throwsA(const TypeMatcher<ProcessRunnerException>()));
+        throwsA(const isInstanceOf<ProcessRunnerException>()));
```
Comment (gspencergoog) [inline dev/bots/test/prepare_package_test.dart]: Did you want to remove the `const` here too?

File: dev/bots/test/prepare_package_test.dart [hunk 2]
```diff
@@ -62,7 +62,7 @@ void main() {
             expectAsync1((List<String> commandLine) async {
               return processRunner.runProcess(commandLine);
             })(<String>['echo', 'test']),
-            throwsA(const TypeMatcher<ProcessRunnerException>()));
+            throwsA(const isInstanceOf<ProcessRunnerException>()));
```
Comment (gspencergoog) [inline dev/bots/test/prepare_package_test.dart]: And here.

File: dev/bots/test/prepare_package_test.dart [hunk 3]
```diff
@@ -204,7 +204,7 @@ void main() {
         };
         processManager.fakeResults = calls;
         expect(expectAsync0(creator.initializeRepo),
-            throwsA(const TypeMatcher<ProcessRunnerException>()));
+            throwsA(const isInstanceOf<ProcessRunnerException>()));
```
Comment (gspencergoog) [inline dev/bots/test/prepare_package_test.dart]: And here.

**Comment by gspencergoog** [comment]
There are a bunch of places where you're still using `const isInstanceOf`, more than just the three I pointed out.  You might want to make another pass to remove the consts, if you want to prepare them for using the function.

**Comment by gspencergoog** [comment]
Ahh, I see.
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Review by gspencergoog — APPROVED** [review]

File: packages/flutter/test/painting/decoration_test.dart [hunk 4]
```diff
@@ -227,10 +227,10 @@ void main() {
     final Invocation call = canvas.invocations.singleWhere((Invocation call) => call.memberName == #drawImageNine);
     expect(call.isMethod, isTrue);
     expect(call.positionalArguments, hasLength(4));
-    expect(call.positionalArguments[0], const TypeMatcher<TestImage>());
+    expect(call.positionalArguments[0], const isInstanceOf<TestImage>());
```
Comment (mravn-google) [inline packages/flutter/test/painting/decoration_test.dart]: We should remove `const` in this file, right?

Comment (Hixie) [reply packages/flutter/test/painting/decoration_test.dart]: That file doesn't use flutter_test, so no.

**Pull request merged**